experimental/packages/otlp-exporter-base/src/platform/browser/OTLPExporterBrowserBase.ts

```diff
@@ -39,6 +40,9 @@ export abstract class OTLPExporterBrowserBase<
     super(config);
     this._useXHR =
       !!config.headers || typeof navigator.sendBeacon !== 'function';
+    if (config.headers && typeof config.headers === 'function') {
+      this._getHeaders = config.headers as () => Record<string, string>;
```

Comment:
If the `Partial<>` are removed from the type declaration, this type cast can be removed as typescript can narrow the type with the condition above.

```suggestion
      this._getHeaders = config.headers;
```

Reply:
Thanks for the tip, I've improved the config.headers type now